a11ym

```diff
@@ -190,19 +188,34 @@ urls.drain = function () {
     }
 };
 
-var crawler             = new Crawler(url.hostname);
-crawler.initialPath     = url.path;
-crawler.initialPort     = url.port || 80;
-crawler.initialProtocol = (url.protocol || 'http').replace(':', '');
-crawler.interval        = 100;
-crawler.maxConcurrency  = 5;
-crawler.maxDepth        = +program.maximumDepth;
+var parseURL = function (url) {
+    var parsed = URL.parse(url);
+
+    if (!parsed.protocol) {
+        parsed.protocol = 'http';
+    } else {
+        parsed.protocol = parsed.protocol.replace(':', '');
+    }
+
+    if (!parsed.port) {
+        parsed.port = 'http' === parsed.protocol ? 80 : 443;
```

Comment:
It seemed to me the crawler added the port to the URL ?

Had some issues with this, but anyway, it makes sense and we can always fix it later if needed !